pkg/dataobj/internal/encoding/decoder.go

```diff
@@ -31,6 +32,10 @@ type (
 		// Columns describes the set of columns in the provided section.
 		Columns(ctx context.Context, section *filemd.SectionInfo) ([]*streamsmd.ColumnDesc, error)
 
+		// SortInfo describes the sort order information for the provided section.
+		// Returns nil if no sort order information is available.
+		SortInfo(ctx context.Context, section *filemd.SectionInfo) ([]*datasetmd.ColumnSortInfo, error)
```

Comment:
Should we squash Columns and SortInfo together into a more general 

```go 
Metadata(ctx context.Context, section *filemd.SectionInfo) (*streamsmd.Metadata, error) 
```

method? The main reason to do this would be to avoid unnecessary roundtrips; right now we _could_ fetch both pieces of metadata in a single request, but the decoder API only permits getting them across two requests. 

(Same comment for LogsDecoder)